packages/main/src/StepInput.ts

```diff
@@ -512,19 +530,53 @@ class StepInput extends UI5Element implements IFormElement {
 		}
 	}
 
+	_isValueWithCorrectPrecision() {
+		// gets either "." or "," as delimiter which is based on locale, and splits the number by it
+		const delimiter = this.input.value.includes(".") ? "." : ",";
+		const numberParts = this.input.value.split(delimiter);
+		const decimalPartLength = numberParts.length > 1 ? numberParts[1].length : 0;
+		const isValueWithCorrectPrecision = decimalPartLength === this.valuePrecision;
+
+		return isValueWithCorrectPrecision;
+	}
+
 	_onInputChange() {
-		if (this.input.value === "") {
-			this.input.value = (this.min || 0) as unknown as string;
+		this._setDefaultInputValueIfNeeded();
+
+		const inputValue = Number(this.input.value);
+		if (this._isValueChanged(inputValue)) {
+			this._updateValueAndValidate(inputValue);
 		}
-		const inputValue = this._preciseValue(parseFloat(this.input.value));
-		if (this.value !== this._previousValue || this.value !== inputValue) {
-			this.value = inputValue;
-			this._validate();
-			this._setButtonState();
-			this._fireChangeEvent();
+	}
+
+	_setDefaultInputValueIfNeeded() {
+		if (this.input.value === "") {
+			const defaultValue = (this.min || 0).toFixed(this.valuePrecision);
+			this.input.value = defaultValue;
```

Comment:
because when the `this.input.value` is empty string the `native input` treats the **empty string** as no value, rather than defaulting to **0**. We decided that we would like to fallback to **0** in case of empty string and round it (to showcase the effect of the `valuePrecision` property).